libraries/SPIFFS/src/SPIFFS.cpp

```diff
@@ -43,21 +43,31 @@ bool SPIFFSImpl::exists(const char* path)
     return (f == true) && !f.isDirectory();
 }
 
-SPIFFSFS::SPIFFSFS() : FS(FSImplPtr(new SPIFFSImpl()))
+SPIFFSFS::SPIFFSFS() : FS(FSImplPtr(new SPIFFSImpl())), partitionLabel_(NULL)
 {
 
 }
 
-bool SPIFFSFS::begin(bool formatOnFail, const char * basePath, uint8_t maxOpenFiles)
+SPIFFSFS::~SPIFFSFS()
 {
-    if(esp_spiffs_mounted(NULL)){
+    if (partitionLabel_){
+        free(partitionLabel_);
+        partitionLabel_ = NULL;
+    }
+}
+
+bool SPIFFSFS::begin(bool formatOnFail, const char * basePath, uint8_t maxOpenFiles, const char * partitionLabel)
+{
+    partitionLabel_ = partitionLabel ? strdup(partitionLabel) : NULL;
```

Comment:
it's maybe possible to call `begin` more than once. In that case make sure that if `partitionLabel_` is not NULL, to free it first :)

Reply:
Done.